src/libsystemd/sd-event/event-util.c

```diff
@@ -0,0 +1,70 @@
+/* SPDX-License-Identifier: LGPL-2.1+ */
+
+#include "event-util.h"
+#include "log.h"
+
+int event_reset_time(sd_event *e, sd_event_source **s, clockid_t clock, uint64_t usec, uint64_t accuracy, sd_event_time_handler_t callback, void *userdata, bool force_reset) {
+        int enabled, r;
+        clockid_t c;
+
+        assert(e);
+        assert(s);
+
+        if (!*s)
+                goto create_new;
+
+        if (!force_reset) {
+                r = sd_event_source_get_enabled(*s, &enabled);
+                if (r < 0) {
+                        log_debug_errno(r, "sd-event: Failed to query whether event source is enabled or not, trying to create new event source: %m");
+                        goto create_new;
```

Comment:
this shouldn't happen if it does, then there's something *really* off. let's just propagate the error here.